spec/system/admin/barcode_items_system_spec.rb

```diff
@@ -24,15 +24,23 @@
 
     it "should edit an existing global barcode"
     it "should delete a global barcode" do
+      table_xpath = '//table[@id="tbl_barcode_items"]'
       visit admin_barcode_items_path
       page.refresh
-      expect(page).to have_content barcode_item.base_item.name
+
+      within(:xpath, table_xpath) do
+        expect(page).to have_content barcode_item.base_item.name
+      end
+
       expect(
         accept_confirm do
           click_on "Delete"
         end
       ).to include "Are you sure you want to delete"
-      expect(page).to have_no_content barcode_item.base_item.name
+
+      within(:xpath, table_xpath) do
+        expect(page).not_to have_content barcode_item.base_item.name
```

Comment:
```suggestion
       expect(page).not_to have_content barcode_item.base_item.name, exact: true
```

I think @mdworken said that the issue was false positives with an item like "Swimmers" showing up in "Cloth Swimmers".  If you use `exact_text: true` here it'll ensure it's matching it correctly.

Reply:
That would probably work in this case, but wouldn't we want to limit the target more precisely to reduce the false positive risk that the text might appear somewhere in the page other than where we're really looking, to reduce the risk that we will need go through this process again? 

I omitted the `exact: true` because I found online help saying that `true` was the default value, and the tests worked. However, I am adding that now and will push momentarily.

Reply:
I've removed the XPath scope, and the `exact: true` is not working. I think exact means that _both_ strings cannot be substrings of each other, i.e. they must be equal. Here is the error message:

```
expected to find exact text "Adult Cloth Diapers (Large/XL/XXL)" in "Toggle navigation\nBack to Dashboard\nDEFAULT SUPERADMIN\nMain Navigation\nAdmin Dashboard\nBarcode Items\nAll Barcodes\nNew Barcoded Item\nBase Items\nOrganizations\nPartners\nUsers\nUser Feedback\nBarcode Items - Global\nAdmin Barcode Items\nFilter by base item category\nAdult Briefs (Large/X-Large)\nAdult Briefs (Medium/Large)\nAdult Briefs (Small/Medium)\nAdult Briefs (XS/Small)\nAdult Briefs (XXL)\nAdult Briefs (XXS)\nAdult Briefs (XXXL)\nAdult Cloth Diapers (Large/XL/XXL)\nAdult Cloth Diapers (Small/Medium)\nAdult Incontinence Pads\nAdult Liners\nBed Pads (Cloth)\nBed Pads (Disposable)\nBibs (Adult & Child)\nCloth Diapers (AIO's/Pocket)\nCloth Diapers (Covers)\nCloth Diapers (Plastic Cover Pants)\nCloth Diapers (Prefolds & Fitted)\nCloth Inserts (For Cloth Diapers)\nCloth Potty Training Pants/Underwear\nCloth Swimmers (Kids)\nDiaper Rash Cream/Powder\nDisposable Inserts\nKids (Newborn)\nKids (Preemie)\nKids (Size 1)\nKids (Size 2)\nKids (Size 3)\nKids (Size 4)\nKids (Size 5)\nKids (Size 6)\nKids (Size 7)\nKids L/XL (60-125 lbs)\nKids Pull-Ups (2T-3T)\nKids Pull-Ups (3T-4T)\nKids Pull-Ups (4T-5T)\nKids S/M (38-65 lbs)\nOther\nPads\nSwimmers\nTampons\nUnderpads (Pack)\nWipes (Adult)\nWipes (Baby)\nFilter Clear Filters\nAdd New Barcode\nItem Type Quantity in the Box Barcode\nActions\nAdult Cloth Diapers (Large/XL/XXL) 50 222074763917 View Edit Delete\nDiaperbase was built with by Ruby for Good.". (However, it was found 2 times using a case insensitive search.)
```

Reply:
I think this would work though:

```ruby
      options = page.all('option').map(&:text)
      expect(options).to include(target_name)
```

...and the negation of that for the following test (`...not_to...`).

Reply:
I think the other approach that I used before that worked was to use `page.text.split("\n").include?` for the test. `page.text.split("\n")` will return something like this:

```
[
    [ 0] "Toggle navigation",
    [ 1] "Back to Dashboard",
    [ 2] "DEFAULT SUPERADMIN",
    [ 3] "Main Navigation",
    [ 4] "Admin Dashboard",
    [ 5] "Barcode Items",
    [ 6] "All Barcodes",
    [ 7] "New Barcoded Item",
    [ 8] "Base Items",
    [ 9] "Organizations",
    [10] "Partners",
    [11] "Users",
    [12] "User Feedback",
    [13] "Barcode Items - Global",
    [14] "Admin Barcode Items",
    [15] "Filter by base item category",
    [16] "Adult Briefs (Large/X-Large)",
    [17] "Adult Briefs (Medium/Large)",
    [18] "Adult Briefs (Small/Medium)",
    [19] "Adult Briefs (XS/Small)",
    [20] "Adult Briefs (XXL)",
    [21] "Adult Briefs (XXS)",
    [22] "Adult Briefs (XXXL)",
    [23] "Adult Cloth Diapers (Large/XL/XXL)",
    [24] "Adult Cloth Diapers (Small/Medium)",
    [25] "Adult Incontinence Pads",
    [26] "Adult Liners",
    [27] "Bed Pads (Cloth)",
    ...
    [64] "Cloth Diapers (Covers) 50 276762366545 View Edit Delete",
    [65] "Diaperbase was built with by Ruby for Good."
```

I don't like this approach though because it is less precise than seeing if the list of `<option>`'s includes the search text.

Reply:
On further research I realized that the `<option>` elements were _not_ what I wanted, so forget I mentioned it. ;)

Reply:
@mdworken , thank you for pointing me in the direction of CSS. I added a CSS class to the table cells we're needing to compare with, and doing an exact match test on the text values at test time. It's a minuscule code change but unlike some of the other strategies we considered, ignores any values outside of that column of that table, making false positives impossible. These code changes are pushed now.